pkg/apiserver/resthandler.go

```diff
@@ -234,7 +256,7 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope RequestScope, forceWatch
 		opts := *out.(*api.ListOptions)
```

Comment:
I'm not sure I understand this comment - this doesn't prevent other groups having their own ListOptions.
What is happening here is that:
- DecodeParametersInto is decoding url query to "v1.ListOptions" (this can potentially also be different than v1)
- the there list conversion to the internal: "scope.Convertor.ConvertToVersion(versioned, "")"
  (all different ListOptions can provide conversion to the internal api)
- then we have api.ListOptions, so we can cast it.

How you would like to change it?